Algorithms

VeraSimon opened the pull request:
- [x] stock_prices
- [x] recipe_batches
- [x] counting_stairs
- [x] rock_paper_scissors
- [ ] making_change
- [ ] (Stretch) knapsack

VeraSimon committed:
Stock prices
Fixed missing EoF newlines in related files

VeraSimon committed:
Stock prices
Came up with an iterative solution with a slightly dirty initialization

VeraSimon committed:
Recipe batches
Came up with iterative solution

VeraSimon committed:
Climbing stairs
Came up with both a recursive and dynamic solution for this after realizing I was fighting an off by 1 error for a bit

VeraSimon committed:
Climbing stairs
Removed a debug print

BrianARuff reviewed (commented):
A very impressive recipe_batches solution, hardly any value changing going on or any extraneous work on the processor. 100% from me 👍 

Also, the find_max_profit solution you posted is O(n^2). If you run it through python tutor you will visually see what is happening with i and j.

VeraSimon committed:
Rock, paper, scissors
Finished a recursive version of RPS

VeraSimon committed:
Stock prices
Updated Big O notes after comment from Brian and looking at the code a bit more

VeraSimon committed:
Making change
Still WIP. Trying to work out the finer details of this final else statement.

VeraSimon commented:
This final else statement in Making Change is perplexing me. I thought using reduce might be a clever way to tackle it, but I haven't quite wrapped my head around the million examples that use lambda with it, and I'd like to tackle one problem at a time here.

VeraSimon committed:
Making change
Forgot to cache stuff. Commented out debugging prints.

VeraSimon committed:
Making change
Still WIP. Slightly more progress. Tests now fail on 10.
